oaknational/Oak-Web-Application
# feat: lesson data

## Description

- adds queries and fixtures for:
  - teachersKeyStageSubjectUnitLessons.gql
  - teachersLessonOverview.gql
  - teachersLessonOverviewPaths.gql
- updates pages, components, and tests to accomodate new queries
- adds appropriate error handling ... updates `OakError` to include the following code:
  - `curriculum-api/not-found`
  - `curriculum-api/uniqueness-assumption-violated`
  - TODO: need to test that errors are reported and correct status codes are sent with each error
- moves "lesson overview" page to url `/key-stages/ks3/subjects/art/units/painting-95846/lessons/what-is-painting-39473`
  - see this [slack conversation ](https://oaknationalacademy.slack.com/archives/C011QBA2VU5/p1671631006988799?thread_ts=1671617113.661199&cid=C011QBA2VU5) for reasoning
  - have updated the [URLs notion page](https://www.notion.so/oaknationalacademy/URL-structure-0228e2ee4ad44a1cb7c9d66685a915fc#6db999cf02604354ac42110520f6eb85) accordingly
- @jack-skerrett-oak (and devs) the video is accessible by tabbing even when the "expandable content" component is closed. This is not a regression as such -- but it seems the expandable content component needs a fix (separate pr) so that it follows a11y guidelines

## Issue(s)

Fixes #1128

## How to test

**This PR relies on https://github.com/oaknational/Data-Tools/pull/480 having been merged** 

1. Merge https://github.com/oaknational/Data-Tools/pull/480
2. Rebuild this PR
3. Go to {cloud link}
4. Click through the navigation to view a lesson
5. You should see actual data coming through

## Checklist
